File: explore.js
var info = [[['6', 'Maxwell Zen', 'https://lh3.googleusercontent.com/a-/AOh14GhKx05Ufson8VKJ_uvFw6wvA0ik0oPG3MbhqgKg2w=s96-c', 'asdf;lkj', 'mzen20@stuy.edu'], ['1', 'Xiaoshen Ma', 'https://www.hobbydb.com/processed_uploads/subject_photo/subject_photo/image/30900/1504725344-16834-3961/CARL_large.jpeg', 'the sweets were very pog champ', 'xma20@stuy.edu']], [['3', 'Maxwell Zen', 'https://lh3.googleusercontent.com/a-/AOh14GhKx05Ufson8VKJ_uvFw6wvA0ik0oPG3MbhqgKg2w=s96-c', 'This is a new post in order to test the extent of my overwhelming sexiness', 'mzen20@stuy.edu'], ['2', 'Jennifer Sun', 'https://www.usajacket.com/blog/wp-content/uploads/2017/07/Gru-Costume-810x427.jpg', 'the pizza was very pog champ', 'jsun20@stuy.edu']], [['5', 'Maxwell Zen', 'https://lh3.googleusercontent.com/a-/AOh14GhKx05Ufson8VKJ_uvFw6wvA0ik0oPG3MbhqgKg2w=s96-c', 'testing testing 1 2 3 xiaoshen got more sleep than me', 'mzen20@stuy.edu'], ['4', 'Maxwell Zen', 'https://lh3.googleusercontent.com/a-/AOh14GhKx05Ufson8VKJ_uvFw6wvA0ik0oPG3MbhqgKg2w=s96-c', 'This is the power of my immense reservoir of brain cells', 'mzen20@stuy.edu'], ['0', 'Maxwell Zen', 'https://i.pinimg.com/originals/bd/a4/04/bda404f24a6471f3319e190bc99e3b91.jpg', 'the tea was very pog champ', 'mzen20@stuy.edu']]];
var pos = [];
for (var i = 0; i < info.length; i++) {
    pos.push(0);
}
var light = 0;
function nxt(r) {
    pos[r] += 1;
    if (pos[r]==info[r].length) {
        pos[r]=0;
    }
    document.getElementById("name" + r.toString()).innerHTML = info[r][pos[r]][1];
    document.getElementById("pfp" + r.toString()).src = info[r][pos[r]][2];
    document.getElementById("message" + r.toString()).innerHTML = info[r][pos[r]][3];
    document.getElementById("nxt" + r.toString()).innerHTML = "Next Story " + (pos[r]+1).toString() + "/" + info[r].length.toString();
}

function toggle() {
    light += 1;
    document.getElementById("aroundmap").innerHTML="<div id='mapid'></div>";
    create();
}

function onSignIn(googleUser) {
  // Useful data for your client-side scripts:
  var profile = googleUser.getBasicProfile();
  document.getElementById('profileinfo').innerHTML = "<br><a>" + profile.getName() + "</a>" + "<a href='mailto:'" + profile.getEmail() + "'>" + profile.getEmail() + "</a><br><img style='margin-left: 32px;' src='" + profile.getImageUrl() + "'/>";
  document.getElementById('signin').style.display="none";
  document.getElementById('signout').style.display="block";
  // document.location = "index.html";
}
function signOut() {
  var auth2 = gapi.auth2.getAuthInstance();
  auth2.signOut().then(function () {
    document.location = "login.html";
  });
}

/* Set the width of the sidebar to 250px and the left margin of the page content to 250px */
function openNav() {
  document.getElementById("mySidebar").style.width = "250px";
  document.getElementById("main").style.marginLeft = "250px";
}

/* Set the width of the sidebar to 0 and the left margin of the page content to 0 */
function closeNav() {
  document.getElementById("mySidebar").style.width = "0";
  document.getElementById("main").style.marginLeft = "0";
}

function create() {
    var myIcon = L.icon({
        iconUrl: 'https://images.vexels.com/media/users/3/185225/isolated/preview/cc21d796204923937547f509844f7eba-hamburger-icon-hamburger-by-vexels.png',
        iconSize: [50, 60],
        iconAnchor: [25, 30],
//

//
    })
    var mymap = L.map('mapid').setView([40.7, -74], 10);
    if (light%2==0) {
        L.tileLayer("https://api.mapbox.com/styles/v1/mapbox/streets-v11/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoieGlhb3BpZXMiLCJhIjoiY2tsZW4zcnk1MWxyaTJvcWVrb2ZuZWRjaSJ9.oehEvW7g2zi8R69mZhanrA", {
        attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, Imagery © <a href="https://www.mapbox.com/">Mapbox</a>',
        maxZoom: 18,
        id: 'mapbox/streets-v11',
        tileSize: 512,
        zoomOffset: -1,
        accessToken: 'pk.eyJ1IjoieGlhb3BpZXMiLCJhIjoiY2tsZW4zcnk1MWxyaTJvcWVrb2ZuZWRjaSJ9.oehEvW7g2zi8R69mZhanrA'
        }).addTo(mymap);
    } else {
        L.tileLayer("https://api.mapbox.com/styles/v1/xiaopies/cklfbn8nn3l9217p29ksw65el/tiles/256/{z}/{x}/{y}@2x?access_token=pk.eyJ1IjoieGlhb3BpZXMiLCJhIjoiY2tsZW4zcnk1MWxyaTJvcWVrb2ZuZWRjaSJ9.oehEvW7g2zi8R69mZhanrA", {
        attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, Imagery © <a href="https://www.mapbox.com/">Mapbox</a>',
        maxZoom: 18,
        // id: 'mapbox/streets-v11',
        tileSize: 512,
        zoomOffset: -1,
        // accessToken: 'pk.eyJ1IjoieGlhb3BpZXMiLCJhIjoiY2tsZW4zcnk1MWxyaTJvcWVrb2ZuZWRjaSJ9.oehEvW7g2zi8R69mZhanrA'
        }).addTo(mymap);
    }

    /*markers*/
}

create();

File: explore.py
#! /usr/bin/python
print('Content-type: text/html\n')
import cgi, cgitb, re
cgitb.enable()
stuff = cgi.FieldStorage()
r = open("resources/restaurants.txt","r").read().strip().split("\n123456789987654321\n")
rc = open("resources/rcoords.txt","r").read().strip().split("\n123456789987654321\n")
for i in range(len(rc)):
    rc[i] = rc[i].split('\n')
s = open("resources/stories.txt","r").read().strip().split("\n123456789987654321\n")
s2 = []
#rest number name pfp message email
for i in range(1, len(s), 6):
    if i > len(s)-5:
        break
    s2 += [[int(s[i+3]),-int(s[i-1]),s[i+1],s[i],s[i+4],s[i+2]]]
s2.sort()
info = []
for i in range(len(r)):
    info += [[]]
for i in s2:
    info[i[0]].append([str(-1 * i[1])] + i[2:])

js = open("explore.js", "r").read()
js = "var info = " + str(info) + ';' + js[js.find('\n'):]


# html = open("explore.html","r").read()

marker = '''var marker = L.marker([/*x*/, /*y*/],{
        title: "Click me!",
        icon: myIcon,
    }).addTo(mymap)
    .bindPopup("<head><h1 class='restaurant' id = 'restaurant<!--r-->'> RESTAURANT </h1> <!--s--> <h3 <!--h--> class='name' id = 'name<!--r-->'> NAME </h3> </head> <body> <img src='PFP' class='pfp' id = 'pfp<!--r-->' <!--h-->><p class='message' id = 'message<!--r-->' <!--h-->> STORY </p> <button class='nxt' onclick = 'nxt(<!--r-->)' id='nxt<!--r-->' <!--h-->> Next Story <!--l--> </button> </body>")
    .openPopup
    '''
toadd = ""
for i in range(len(r)):
    copy = marker
    copy = copy.replace("/*x*/", rc[i][0])
    copy = copy.replace("/*y*/", rc[i][1])
    copy = copy.replace("<!--r-->", str(i))
    copy = copy.replace("RESTAURANT", r[i])
    if len(info[i])==0:
        copy = copy.replace("<!--h-->", "style='display: none;'")
        copy = copy.replace("<!--s-->", "<h3 class='nomessage'> No Stories </h3>")
    else:
        copy = copy.replace("<!--h-->", "")
        copy = copy.replace("<!--s-->", "")
        copy = copy.replace("NAME", info[i][0][1])
        copy = copy.replace("PFP", info[i][0][2])
        copy = copy.replace("STORY", info[i][0][3])
        copy = copy.replace("<!--l-->", "1/" + str(len(info[i])))
    toadd += copy
js = js.replace("/*markers*/", toadd)
html = open('explore.html','r').read()
html = html.replace("<!--script-->", "<script>" + js + "</script>")
print(html)
